File: app/src/main/java/com/github/anrimian/musicplayer/di/app/library/albums/AlbumsComponent.java
package com.github.anrimian.musicplayer.di.app.library.albums;

import com.github.anrimian.musicplayer.di.app.library.albums.items.AlbumItemsComponent;
import com.github.anrimian.musicplayer.di.app.library.albums.items.AlbumItemsModule;
import com.github.anrimian.musicplayer.ui.library.albums.list.AlbumsListPresenter;

import dagger.Subcomponent;

@Subcomponent(modules = AlbumsModule.class)
public interface AlbumsComponent {

    AlbumItemsComponent albumItemsComponent(AlbumItemsModule module);

    AlbumsListPresenter albumsListPresenter();
}

File: app/src/main/java/com/github/anrimian/musicplayer/di/app/library/albums/AlbumsComponent.kt
package com.github.anrimian.musicplayer.di.app.library.albums

import com.github.anrimian.musicplayer.di.app.library.albums.items.AlbumItemsComponent
import com.github.anrimian.musicplayer.di.app.library.albums.items.AlbumItemsModule
import com.github.anrimian.musicplayer.ui.library.albums.list.AlbumsListPresenter
import dagger.Subcomponent

@Subcomponent(modules = [AlbumsModule::class])
interface AlbumsComponent {

    fun albumItemsComponent(module: AlbumItemsModule): AlbumItemsComponent

    fun albumsListPresenter(): AlbumsListPresenter

}